### ISSUE TITLE
UPI Payment is not showing in Android level 11 & 10.

### ISSUE THREAD

**Issue by altruistindia** [PatilShreyas/EasyUpiPayment-Android#149]
Write information about Bug/Feature Request here
UPI payment option is not showing in android level 10 and 11, but it showing in lower level

**Comment by sharmaAshish-dev** [comment]
When an app targets Android 11 (API level 30) or higher and queries for information about the other apps that are installed on a device, the system filters this information by default. The limited package visibility reduces the number of apps that appear to be installed on a device, from your app's perspective.

Solution: 
you have to add the following code in your manifest file anywhere outside <application> tag.

<queries>
        <intent>
            <action android:name="android.intent.action.VIEW" />
            <data android:scheme="upi"/>
        </intent>
</queries>

**Comment by Rajkumar9898** [comment]
> When an app targets Android 11 (API level 30) or higher and queries for information about the other apps that are installed on a device, the system filters this information by default. The limited package visibility reduces the number of apps that appear to be installed on a device, from your app's perspective.
> 
> Solution: you have to add the following code in your manifest file anywhere outside tag.

Above you have mention the code but you have not attached the code please provide it.

**Comment by Rajkumar9898** [comment]
I found the solution for it which is  below,
put this code in manifest file and it will work fine.
Thanks.

  <queries>
        
        <intent>
            <action android:name="android.intent.action.VIEW" />
            <data android:scheme="upi" />
        </intent>

        
        <intent>
            <action android:name="android.intent.action.CHOOSER" />
            <data android:scheme="upi" />
        </intent>

        
        <package android:name="in.amazon.mShop.android.shopping" />
        <package android:name="in.org.npci.upiapp" />
        <package android:name="com.google.android.apps.nbu.paisa.user" />
        <package android:name="com.phonepe.app" />
        <package android:name="net.one97.paytm" />
    </queries>

**Comment by sharmaAshish-dev** [comment]
> 

Yes, indeed it is. but only Intent filter can also do the job done. As it will also allows your app to query all apps that are provoked by "upi Scheme" rather than querying specific packages.👍
